test/unit/org/apache/cassandra/cql3/validation/operations/UseTest.java

```diff
@@ -19,13 +19,38 @@
 
 import org.junit.Test;
 
+import org.apache.cassandra.config.DatabaseDescriptor;
 import org.apache.cassandra.cql3.CQLTester;
+import org.apache.cassandra.cql3.QueryProcessor;
+import org.apache.cassandra.exceptions.InvalidRequestException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
 
 public class UseTest extends CQLTester
 {
     @Test
     public void testUseStatementWithBindVariable() throws Throwable
     {
+        DatabaseDescriptor.setUseStatementsEnabled(true);
         assertInvalidSyntaxMessage("Bind variables cannot be used for keyspace names", "USE ?");
     }
+
+    @Test
+    public void shouldRejectUseStatementWhenProhibited() throws Throwable
+    {
+        long useCountBefore = QueryProcessor.metrics.useStatementsExecuted.getCount();
+
+        try
+        {
+            DatabaseDescriptor.setUseStatementsEnabled(false);
+            execute("USE cql_test_keyspace");
+            fail("expected USE statement to fail with use_statements_enabled = false");
+        }
+        catch (InvalidRequestException e)
+        {
+            assertEquals(useCountBefore, QueryProcessor.metrics.useStatementsExecuted.getCount());
+            assertMessageContains("USE statements prohibited", e);
```

Comment:
nit: FYI, you can use `Assertions.assertThat(e).hasMessageContaining("USE statements prohibited");`, does the same thing, just without exposing the hidden method